src/stream/Encryption.ts

```diff
@@ -220,7 +293,7 @@ class EncryptionUtilBase {
      * message content and returns null.
      */
 
-    static decryptStreamMessage(streamMessage, groupKey) {
+    static decryptStreamMessage(streamMessage: MessageLayer.StreamMessage, groupKey: GroupKey) {
```

Comment:
The static methods which take a non-null `groupKey` as a parameter could be instance methods of `GroupKey` class? That way this this `EncryptionUtilBase` would focus on abstract publicKey/privateKey handling.

Reply:
Another leftover from previous implementation.

They could definitely live elsewhere agreed, though not sure if they belong on `GroupKey`. This logic is implementation detail for the stream message encryption/decryption process, rather than anything specific to the `GroupKey` data. Adding this to GroupKey would make `GroupKey` aware of `StreamMessage`. 

I'd prefer to have the `GroupKey` only knowing about `GroupKey` stuff, keep it as close to a self-contained data container as possible avoid slipping into "Fat Model" architecture. i.e. attach cross-model logic to features rather than datatypes.

Reply:
Yeah, not good to couple components together too much.

One possibility would be to create a separate `StreamMessageEncryptor` which would know `StreamMessage` and `GroupKey`. 

Or maybe that method could be in `StreamMessage`, if we create an encryptor interface. `StreamMessage` could know that, but not the actual implementing `GroupKeyEncryptor`.